server/src/game.rs

```diff
@@ -335,6 +337,20 @@ impl Game {
     pub fn on_item_collect(&mut self, world: &mut World, event: ItemCollectEvent) {
         on_item_collect_broadcast(self, world, &event);
     }
+
+    /// Called when weather changes
+    pub fn on_weather_change(&mut self, world: &mut World, event: &mut WeatherChangeEvent) {
```

Comment:
I think we also need to send weather to a player when they join, or am I missing something?